ironfish/src/wallet/wallet.ts

```diff
@@ -1433,12 +1436,26 @@ export class Wallet {
         publicKeyPackage: accountValue.multisigKeys.publicKeyPackage,
         secret: multisigIdentity.secret.toString('hex'),
       }
+      secret = multisigIdentity.secret
+    }
+
+    if (accountValue.multisigKeys && isMultisigSignerImport(accountValue.multisigKeys)) {
+      secret = Buffer.from(accountValue.multisigKeys.secret, 'hex')
     }
 
     if (name && this.getAccountByName(name)) {
       throw new DuplicateAccountNameError(name)
     }
 
+    if (secret) {
+      const duplicateSecret = await this.walletDb.getMultisigSecretByName(accountValue.name)
+      if (duplicateSecret && duplicateSecret.equals(secret)) {
```

Comment:
should this be `&& !duplicateSecret.equals(secret)`?

Reply:
updated the logic here + test